src/main/java/com/notably/logic/commands/DeleteCommand.java

```diff
@@ -2,49 +2,38 @@
 
 import static java.util.Objects.requireNonNull;
 
-import java.util.List;
-
-import com.notably.commons.core.index.Index;
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.commons.core.path.Path;
+import com.notably.commons.core.path.RelativePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
 import com.notably.logic.commands.exceptions.CommandException;
 import com.notably.model.Model;
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
+ * Represents a command that deletes a block.
  */
 public class DeleteCommand extends Command {
-
     public static final String COMMAND_WORD = "delete";
+    private Path targetPath;
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
-
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
-
-    private final Index targetIndex;
-
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(Path targetPath) {
+        this.targetPath = targetPath;
```

Comment:
I think I remember you saying that you'd like to make `Path` accept both `RelativePath` and `AbsolutePath`. I personally think that the responsibility of knowing how to handle each type should be given to the `DeleteCommandParser` class. You then would only need to worry about one type of `Path` here. In fact, I believe that you only create `DeleteCommand` (from within `DeleteCommandParser`) with a `RelativePath`

Reply:
I'll like the `DeleteCommand` to only accept `AbsolutePath`. But i dont think its possible to change an relativePath to an Absolute one in the `DeleteCommandParser` thus i had to do it in the `DeleteCommand` instead
Oh yes i've made it so that the Parser detects both absolute and relative path. Thank for pointing it out .